components/auth/org.wso2.carbon.auth.rest.api.commons/src/test/java/org/wso2/carbon/auth/rest/api/commons/util/RestApiUtilTest.java

```diff
@@ -106,4 +113,66 @@ public void testGetPaginationParams() throws Exception {
         Assert.assertTrue(resultMap.get(RestApiConstants.PAGINATION_PREVIOUS_OFFSET) == 0L);
         Assert.assertTrue(resultMap.get(RestApiConstants.PAGINATION_PREVIOUS_LIMIT) == 10L);
     }
+
+    @Test
+    public void testGetAllowedOrigin() {
+        Assert.assertEquals(RestApiUtil.getAllowedOrigin("http://localhost:9443"), "http://localhost:9443");
+    }
+
+    @Test
+    public void testGetDefinedMethodHeadersInSwaggerContentForDCRM() throws AuthException, NoSuchMethodException {
+        Method method = TestDCRMService.class.getMethod("getApplication", null);
+        Request request = Mockito.mock(Request.class);
+        Mockito.when(request.getUri()).thenReturn("/api/identity/oauth2/dcr/v1.0/register/abcde-ddd");
+        Assert.assertEquals(RestApiUtil.getDefinedMethodHeadersInSwaggerContent(request, method), "GET, PUT, DELETE");
+    }
+
+    @Test
+    public void testGetDefinedMethodHeadersInSwaggerContentForIntrospect() throws AuthException, NoSuchMethodException {
+        Method method = TestIntrospectService.class.getMethod("introspect");
+        Request request = Mockito.mock(Request.class);
+        Mockito.when(request.getUri()).thenReturn("/api/identity/oauth2/introspect/v1.0/introspect");
+        Assert.assertEquals(RestApiUtil.getDefinedMethodHeadersInSwaggerContent(request, method), "POST");
+    }
+
+    @Test
+    public void testGetDefinedMethodHeadersInSwaggerContentForTokenApi() throws AuthException, NoSuchMethodException {
+        Method method = TestTokenApiService.class.getMethod("tokenPost");
+        Request request = Mockito.mock(Request.class);
+        Mockito.when(request.getUri()).thenReturn("/api/auth/oauth2/v1.0/token");
+        Assert.assertEquals(RestApiUtil.getDefinedMethodHeadersInSwaggerContent(request, method), "POST");
```

Comment:
We need to test for both Options and Post.

Reply:
I think we have to allow OPTIONS anyway and don't need to tell that explicitly.